Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserTreeToTableProxyModel.cpp

```diff
@@ -71,19 +75,50 @@ namespace AzToolsFramework
             return m_tableToTree.remove(row) != 0;
         }
 
-        TreeIterator IndexToMap::Insert(TableType map, TreeType row)
+        void IndexToMap::Reserve(TableMap::size_type size)
         {
-            if (m_treeToTable.contains(map))
+            m_treeToTable.reserve(size);
+        }
+
+        void IndexToMap::Insert(TableType map, TreeType row)
+        {
+            // m_treeToTable maps from ModelIndex to int
+            // m_tableToTree maps from int to ModelIndex
+
+            // we have to erase from both maps before we update them.
+            // avoid using "Remove" on the hash, as that causes a re-hash.  Strategy here
+            // will be to update the value if we find it, and erase the value in the other container.
+            // this will yield the minimum amount of re-hashing and memory clearing.
+            bool insertIntoTableToTree = true;
+            bool insertIntoTreeToTable = true;
+            if (auto it = m_treeToTable.find(map); it != m_treeToTable.end())
             {
-                m_tableToTree.remove(m_treeToTable.take(map));
+                int& currentValue = it.value();
+                auto elementInOtherMap = m_tableToTree.find(currentValue);
+                m_tableToTree.erase(elementInOtherMap);
+                currentValue = row;
+                insertIntoTreeToTable = false;
             }
-            if (m_tableToTree.contains(row))
+
+            if (auto it = m_tableToTree.find(row); it != m_tableToTree.end())
+            {
+                QPersistentModelIndex& currentValue = it.value();
+                auto elementInOtherMap = m_treeToTable.find(currentValue);
+                m_treeToTable.erase(elementInOtherMap);
```

Comment:
This should use the `key_type` overload of `hash_table::erase` or it should check that `elementInOtherMap != m_treeToTable.end()`

```suggestion
                auto elementInOtherMap = m_treeToTable.erase(currentValue);
```

Reply:
These are not AZStd structures
```
using TreeHash = QHash<QPersistentModelIndex, int>;
        using TableMap = QMap<int, QPersistentModelIndex>;
        ``` 
they don't actually have similar operations.  erase, for example, requires an iterator, and remove does not (operates onv alue) - but erase does not re-balance the hash and remove does, etc.

Reply:
My mistake. I thought the containers was an AZStd container.